dlib/cmake_utils/find_blas.cmake

```diff
@@ -194,290 +219,25 @@ if (UNIX OR MINGW)
       /usr/lib/atlas
       /usr/lib/openblas-base
       /opt/OpenBLAS/lib
-      $ENV{OPENBLAS_HOME}/lib
-      )
-
-   if (NOT blas_found)
-      find_library(cblas_lib NAMES openblasp openblas PATHS ${extra_paths})
-      if (cblas_lib)
-         set(blas_libraries ${cblas_lib})
-         set(blas_found 1)
-         message(STATUS "Found OpenBLAS library")
-         set(CMAKE_REQUIRED_LIBRARIES ${blas_libraries})
-         # If you compiled OpenBLAS with LAPACK in it then it should have the
-         # sgetrf_single function in it.  So if we find that function in
-         # OpenBLAS then just use OpenBLAS's LAPACK. 
-         CHECK_FUNCTION_EXISTS(sgetrf_single OPENBLAS_HAS_LAPACK)
-         if (OPENBLAS_HAS_LAPACK)
-            message(STATUS "Using OpenBLAS's built in LAPACK")
-            # set(lapack_libraries gfortran) 
-            set(lapack_found 1)
-         endif()
-      endif()
-      mark_as_advanced( cblas_lib)
-   endif()
-
-
-   if (NOT lapack_found)
-      find_library(lapack_lib NAMES lapack lapack-3 PATHS ${extra_paths})
-      if (lapack_lib)
-         set(lapack_libraries ${lapack_lib})
-         set(lapack_found 1)
-         message(STATUS "Found LAPACK library")
-      endif()
-      mark_as_advanced( lapack_lib)
-   endif()
-
-
-   # try to find some other BLAS libraries if we didn't find the MKL
-
-   if (NOT blas_found)
-      find_library(atlas_lib atlas PATHS ${extra_paths})
-      find_library(cblas_lib cblas PATHS ${extra_paths})
-      if (atlas_lib AND cblas_lib)
-         set(blas_libraries ${atlas_lib} ${cblas_lib})
-         set(blas_found 1)
-         message(STATUS "Found ATLAS BLAS library")
-      endif()
-      mark_as_advanced( atlas_lib cblas_lib)
-   endif()
-
-   # CentOS 7 atlas
-   if (NOT blas_found)
-      find_library(tatlas_lib tatlas PATHS ${extra_paths})
-      find_library(satlas_lib satlas PATHS ${extra_paths})
-      if (tatlas_lib AND satlas_lib )
-         set(blas_libraries ${tatlas_lib} ${satlas_lib})
-         set(blas_found 1)
-         message(STATUS "Found ATLAS BLAS library")
-      endif()
-      mark_as_advanced( tatlas_lib satlas_lib)
-   endif()
-
-
-   if (NOT blas_found)
-      find_library(cblas_lib cblas PATHS ${extra_paths})
-      if (cblas_lib)
-         set(blas_libraries ${cblas_lib})
-         set(blas_found 1)
-         message(STATUS "Found CBLAS library")
-      endif()
-      mark_as_advanced( cblas_lib)
-   endif()
-
-
-   if (NOT blas_found)
-      find_library(generic_blas blas PATHS ${extra_paths})
-      if (generic_blas)
-         set(blas_libraries ${generic_blas})
-         set(blas_found 1)
-         message(STATUS "Found BLAS library")
-      endif()
-      mark_as_advanced( generic_blas)
-   endif()
```

Comment:
Why get rid of these ones?

Reply:
In fact, if you use 
```
set(BLA_PREFER_PKGCONFIG 1)
find_package(BLAS)
```
it will first check with pkg-config then check the rest. I think that's a sufficient check if MKL isn't found. I'll do a commit and test. I'll see if i can get the windows build to use blas.

Reply:
Eh. I would rather not remove stuff that isn't broken.  I have in the past used dlib for instance on systems where those blas libraries were what was available.  It wouldn't blow my mind if there are users out there that rely on those.  Intel MKL and openblas are the most popular right now.  But invariably there are sub-communities of people in their own worlds.

Reply:
The CI/CD tests MKL (of course) and openblas and that's all happy. I've tested with atlas as well and it works on my machine. We could add more jobs to Github Actions if you want. Probs a good idea anyway. In fact, the unit tests haven't been working with MKL for some time now which shows how many people are actually properly testing dlib... With the updated `build_cpp.yml` it's super easy to add more configurations.  The default [FindBLAS](https://cmake.org/cmake/help/latest/module/FindBLAS.html) is a pretty good catch all. We can add back some of those fallbacks but there's quite a lot of code there to cater for maybe a 1 or 2 users world-wide. We should maybe embrace the KISS principle a bit more for this kind of stuff.

Reply:
Added ATLAS job

Reply:
Yeah alright, we can see what the default FindBLAS is like.  That stuff used to not be great but maybe it's dope now.